diff -urpN boost_1_48_0/libs/math/config/CMakeLists.txt boost_1_48_0.pm/libs/math/config/CMakeLists.txt
--- boost_1_48_0/libs/math/config/CMakeLists.txt	1969-12-31 19:00:00.000000000 -0500
+++ boost_1_48_0.pm/libs/math/config/CMakeLists.txt	2012-01-25 06:13:29.000000000 -0500
@@ -0,0 +1,3 @@
+project(has_long_double_support CXX)
+add_executable(has_long_double_support has_long_double_support.cpp)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../../../)
diff -urpN boost_1_48_0/libs/math/src/tr1/CMakeLists.txt boost_1_48_0.pm/libs/math/src/tr1/CMakeLists.txt
--- boost_1_48_0/libs/math/src/tr1/CMakeLists.txt	2012-01-25 07:32:50.000000000 -0500
+++ boost_1_48_0.pm/libs/math/src/tr1/CMakeLists.txt	2012-01-25 07:29:01.000000000 -0500
@@ -51,7 +51,8 @@ boost_add_library(math_c99f
 
 try_compile(HAS_LONG_DOUBLE_SUPPORT
 	${CMAKE_CURRENT_SOURCE_DIR}/../../config/
-	${CMAKE_CURRENT_SOURCE_DIR}/../../config/has_long_double_support.cpp)
+	${CMAKE_CURRENT_SOURCE_DIR}/../../config/
+	has_long_double_support)
 
 IF (HAS_LONG_DOUBLE_SUPPORT)
 	boost_add_library(math_tr1l
