diff -up boost_1_37_0/configure.smp boost_1_37_0/configure
--- boost_1_37_0/configure.smp	2009-02-05 18:49:01.000000000 +0100
+++ boost_1_37_0/configure	2009-02-05 18:53:55.000000000 +0100
@@ -347,7 +347,7 @@ LIBS=$LIBS
 
 all: .dummy
 	@echo "\$(BJAM) \$(BJAM_CONFIG) --user-config=user-config.jam \$(LIBS) stage"
-	@\$(BJAM) \$(BJAM_CONFIG) --user-config=user-config.jam \$(LIBS) stage || \\
+	@\$(BJAM) \$(BJAM_CONFIG) --user-config=user-config.jam \$(LIBS) !!!SMP_FLAGS!!! stage || \\
 	echo "Not all Boost libraries built properly."
 
 clean: .dummy
