*** tools/build/v2/tools/gcc.jam.orig	2009-05-05 12:45:31.000000000 -0700
--- tools/build/v2/tools/gcc.jam	2009-05-05 12:48:23.000000000 -0700
*************** if [ os.name ] != NT && [ os.name ] != O
*** 355,361 ****
      # OSF does have an option called -soname but it does not seem to work as
      # expected, therefore it has been disabled.
      HAVE_SONAME   = "" ;
!     SONAME_OPTION = -h ;
  }
  
  
--- 355,362 ----
      # OSF does have an option called -soname but it does not seem to work as
      # expected, therefore it has been disabled.
      HAVE_SONAME   = "" ;
!     SONAME_OPTION = -soname ;
!     SONAME_VERSION = _FEDORA_SONAME ;
  }
  
  
*************** rule link.dll ( targets * : sources * : 
*** 793,799 ****
  # Differs from 'link' above only by -shared.
  actions link.dll bind LIBRARIES
  {
!     "$(CONFIG_COMMAND)" -L"$(LINKPATH)" -Wl,$(RPATH_OPTION:E=-R)$(SPACE)-Wl,"$(RPATH)" "$(.IMPLIB-COMMAND)$(<[1])" -o "$(<[-1])" $(HAVE_SONAME)-Wl,$(SONAME_OPTION)$(SPACE)-Wl,$(<[-1]:D=) -shared $(START-GROUP) "$(>)" "$(LIBRARIES)" $(FINDLIBS-ST-PFX) -l$(FINDLIBS-ST) $(FINDLIBS-SA-PFX) -l$(FINDLIBS-SA) $(END-GROUP) $(OPTIONS) $(USER_OPTIONS)
  }
  
  rule setup-threading ( targets * : sources * : properties * )
--- 794,800 ----
  # Differs from 'link' above only by -shared.
  actions link.dll bind LIBRARIES
  {
!     "$(CONFIG_COMMAND)" -L"$(LINKPATH)" -Wl,$(RPATH_OPTION:E=-R)$(SPACE)-Wl,"$(RPATH)" "$(.IMPLIB-COMMAND)$(<[1])" -o "$(<[-1])" $(HAVE_SONAME)-Wl,$(SONAME_OPTION)$(SPACE)-Wl,$(<[-1]:D=).$(SONAME_VERSION) -shared $(START-GROUP) "$(>)" "$(LIBRARIES)" $(FINDLIBS-ST-PFX) -l$(FINDLIBS-ST) $(FINDLIBS-SA-PFX) -l$(FINDLIBS-SA) $(END-GROUP) $(OPTIONS) $(USER_OPTIONS)
  }
  
  rule setup-threading ( targets * : sources * : properties * )
