*** Jamroot.orig	2009-05-06 12:46:31.000000000 -0700
--- Jamroot	2009-05-06 18:18:17.000000000 -0700
*************** rule tag ( name : type ? : property-set 
*** 344,350 ****
          else
          {
              local result = [ common.format-name
!                 <base> 
                  -$(BUILD_ID)
                  : $(name) : $(type) : $(property-set) ] ;
  
--- 344,350 ----
          else
          {
              local result = [ common.format-name
!                 <base> <threading>
                  -$(BUILD_ID)
                  : $(name) : $(type) : $(property-set) ] ;
  
*************** rule tag ( name : type ? : property-set 
*** 356,362 ****
              # suffixes either. Pgi compilers can not accept library with version
              # suffix.
              if $(type) = SHARED_LIB &&
!               ( ! ( [ $(property-set).get <target-os> ] in windows cygwin darwin aix ) &&
                  ! ( [ $(property-set).get <toolset> ] in pgi ) )
              {
                  result = $(result).$(BOOST_VERSION)  ;
--- 356,362 ----
              # suffixes either. Pgi compilers can not accept library with version
              # suffix.
              if $(type) = SHARED_LIB &&
!               ( ! ( [ $(property-set).get <target-os> ] in windows cygwin darwin aix linux ) &&
                  ! ( [ $(property-set).get <toolset> ] in pgi ) )
              {
                  result = $(result).$(BOOST_VERSION)  ;
