*** boost/config/compiler/gcc.hpp.orig	2003-09-08 09:58:36.000000000 -0500
--- boost/config/compiler/gcc.hpp	2003-09-08 09:59:30.000000000 -0500
***************
*** 48,54 ****
  // those platforms where we can know for sure). It will get turned off again
  // later if no threading API is detected.
  //
! #if !defined(__MINGW32__) && !defined(__CYGWIN__) && !defined(linux) && !defined(__linux) && !defined(__linux__)
  # define BOOST_HAS_THREADS
  #endif 
  
--- 48,54 ----
  // those platforms where we can know for sure). It will get turned off again
  // later if no threading API is detected.
  //
! #if !defined(__MINGW32__) && !defined(__CYGWIN__)
  # define BOOST_HAS_THREADS
  #endif 
  
