diff -urp boost_1_36_0_beta1-orig/tools/build/v2/tools/gcc.jam boost_1_36_0_beta1/tools/build/v2/tools/gcc.jam
--- boost_1_36_0_beta1-orig/tools/build/v2/tools/gcc.jam	2008-08-11 13:58:22.000000000 +0200
+++ boost_1_36_0_beta1/tools/build/v2/tools/gcc.jam	2008-08-11 13:59:21.000000000 +0200
@@ -288,7 +288,7 @@ flags gcc.compile PCH_FILE <pch>on : <pc
 
 # Declare flags and action for compilation
 flags gcc.compile OPTIONS <optimization>off   : -O0 ;
-flags gcc.compile OPTIONS <optimization>speed : -O3 ;
+flags gcc.compile OPTIONS <optimization>speed : "$RPM_OPT_FLAGS" ;
 flags gcc.compile OPTIONS <optimization>space : -Os ;
 
 flags gcc.compile OPTIONS <inlining>off  : -fno-inline ;
Only in boost_1_36_0_beta1/tools/build/v2/tools: gcc.jam~
