*** tools/build/v1/boost-base.jam.orig	2005-03-15 22:38:05.110845776 -0600
--- tools/build/v1/boost-base.jam	2005-03-15 22:39:21.250352319 -0600
*************** rule common-variant-tag ( toolset varian
*** 2651,2660 ****
          version-tag = $(version-tag:J="_") ;
      }
      
!     tags += $(toolset-tag:J=) ;
      tags += $(thread-tag:J=) ;
!     tags += $(runtime-tag:J=) ;
!     tags += $(version-tag) ;
      
      if $(tags)
      {
--- 2651,2660 ----
          version-tag = $(version-tag:J="_") ;
      }
      
! #    tags += $(toolset-tag:J=) ;
      tags += $(thread-tag:J=) ;
! #    tags += $(runtime-tag:J=) ;
! #    tags += $(version-tag) ;
      
      if $(tags)
      {
