*** tools/build/v1/gcc-tools.jam	2004-11-29 17:38:40.295847046 -0600
--- tools/build/v1/gcc-tools.jam.new	2004-11-29 17:39:37.962664115 -0600
*************** flags gcc LINKFLAGS <runtime-link>static
*** 60,66 ****
  flags gcc CFLAGS <debug-symbols>on : -g ;
  flags gcc LINKFLAGS <debug-symbols>on : -g ;
  flags gcc CFLAGS <optimization>off : -O0 ;
! flags gcc CFLAGS <optimization>speed : -O3 ;
  
  # Other optimizations we might want for GCC
  # -fforce-mem -fomit-frame-pointer
--- 60,66 ----
  flags gcc CFLAGS <debug-symbols>on : -g ;
  flags gcc LINKFLAGS <debug-symbols>on : -g ;
  flags gcc CFLAGS <optimization>off : -O0 ;
! flags gcc CFLAGS <optimization>speed : -O2 ;
  
  # Other optimizations we might want for GCC
  # -fforce-mem -fomit-frame-pointer
*************** rule Link-action
*** 387,393 ****
      if $(DLLVERSION) && $(TARGET_TYPE) in $(SHARED_TYPES) && ( $(OS) = LINUX || $(OS) = OPENBSD ) && ! $(NO_GNU_LN)
      {
          OUTTAG on $(<) = ".$(DLLVERSION)" ;
!         SOTAG on $(<) = ".$(DLLVERSION)" ;
          ACTION_1 on $(<) = "" ;
      }
      else
--- 387,393 ----
      if $(DLLVERSION) && $(TARGET_TYPE) in $(SHARED_TYPES) && ( $(OS) = LINUX || $(OS) = OPENBSD ) && ! $(NO_GNU_LN)
      {
          OUTTAG on $(<) = ".$(DLLVERSION)" ;
!         SOTAG on $(<) = "" ;
          ACTION_1 on $(<) = "" ;
      }
      else
