*** boost/config/compiler/gcc.hpp.orig	2005-08-17 15:15:01.000000000 -0500
--- boost/config/compiler/gcc.hpp	2005-08-17 15:15:19.000000000 -0500
***************
*** 91,98 ****
  #  error "Compiler not configured - please reconfigure"
  #endif
  //
! // last known and checked version is 4.0 (Pre-release):
! #if (__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ > 0))
  #  if defined(BOOST_ASSERT_CONFIG)
  #     error "Unknown compiler version - please run the configure tests and report the results"
  #  else
--- 91,98 ----
  #  error "Compiler not configured - please reconfigure"
  #endif
  //
! // last known and checked version is 4.1 (Pre-release):
! #if (__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ > 1))
  #  if defined(BOOST_ASSERT_CONFIG)
  #     error "Unknown compiler version - please run the configure tests and report the results"
  #  else
