diff -urp boost_1_36_0_beta1-orig/boost/archive/polymorphic_iarchive.hpp boost_1_36_0_beta1/boost/archive/polymorphic_iarchive.hpp
--- boost_1_36_0_beta1-orig/boost/archive/polymorphic_iarchive.hpp	2008-08-11 23:49:44.000000000 +0200
+++ boost_1_36_0_beta1/boost/archive/polymorphic_iarchive.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -35,6 +35,8 @@ namespace std{
 #include <boost/serialization/nvp.hpp>
 #include <boost/archive/detail/register_archive.hpp>
 
+#include <limits.h>
+
 // determine if its necessary to handle (u)int64_t specifically
 // i.e. that its not a synonym for (unsigned) long
 // if there is no 64 bit int or if its the same as a long
Only in boost_1_36_0_beta1/boost/archive: polymorphic_iarchive.hpp.orig
diff -urp boost_1_36_0_beta1-orig/boost/archive/polymorphic_oarchive.hpp boost_1_36_0_beta1/boost/archive/polymorphic_oarchive.hpp
--- boost_1_36_0_beta1-orig/boost/archive/polymorphic_oarchive.hpp	2008-08-11 23:49:44.000000000 +0200
+++ boost_1_36_0_beta1/boost/archive/polymorphic_oarchive.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -34,6 +34,8 @@ namespace std{
 #include <boost/serialization/nvp.hpp>
 #include <boost/archive/detail/register_archive.hpp>
 
+#include <limits.h>
+
 // determine if its necessary to handle (u)int64_t specifically
 // i.e. that its not a synonym for (unsigned) long
 // if there is no 64 bit int or if its the same as a long
Only in boost_1_36_0_beta1/boost/archive: polymorphic_oarchive.hpp.orig
diff -urp boost_1_36_0_beta1-orig/boost/date_time/tz_db_base.hpp boost_1_36_0_beta1/boost/date_time/tz_db_base.hpp
--- boost_1_36_0_beta1-orig/boost/date_time/tz_db_base.hpp	2008-08-11 23:49:45.000000000 +0200
+++ boost_1_36_0_beta1/boost/date_time/tz_db_base.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -158,7 +158,7 @@ namespace boost {
       typedef typename time_zone_type::base_type time_zone_base_type;
       typedef typename time_zone_type::time_duration_type time_duration_type;
       typedef time_zone_names_base<char_type> time_zone_names;
-      typedef dst_adjustment_offsets<time_duration_type> dst_adjustment_offsets;
+      typedef dst_adjustment_offsets<time_duration_type> dst_adjustment_offsets_t;
       typedef std::basic_string<char_type> string_type;
 
       //! Constructs an empty database
@@ -346,18 +346,18 @@ namespace boost {
         time_duration_type utc_offset = 
           str_from_delimited_time_duration<time_duration_type,char_type>(result[GMTOFFSET]);
         
-        dst_adjustment_offsets adjust(time_duration_type(0,0,0),
-                                      time_duration_type(0,0,0),
-                                      time_duration_type(0,0,0));
+        dst_adjustment_offsets_t adjust(time_duration_type(0,0,0),
+					time_duration_type(0,0,0),
+					time_duration_type(0,0,0));
 
         boost::shared_ptr<rule_type> rules;
 
         if(has_dst){
-          adjust = dst_adjustment_offsets(
-                                          str_from_delimited_time_duration<time_duration_type,char_type>(result[DSTADJUST]),
-                                          str_from_delimited_time_duration<time_duration_type,char_type>(result[START_TIME]),
-                                          str_from_delimited_time_duration<time_duration_type,char_type>(result[END_TIME])
-                                          );
+          adjust = dst_adjustment_offsets_t(
+					    str_from_delimited_time_duration<time_duration_type,char_type>(result[DSTADJUST]),
+					    str_from_delimited_time_duration<time_duration_type,char_type>(result[START_TIME]),
+					    str_from_delimited_time_duration<time_duration_type,char_type>(result[END_TIME])
+					    );
 
           rules = 
             boost::shared_ptr<rule_type>(parse_rules(result[START_DATE_RULE],
Only in boost_1_36_0_beta1/boost/date_time: tz_db_base.hpp.orig
diff -urp boost_1_36_0_beta1-orig/boost/regex/v4/basic_regex_creator.hpp boost_1_36_0_beta1/boost/regex/v4/basic_regex_creator.hpp
--- boost_1_36_0_beta1-orig/boost/regex/v4/basic_regex_creator.hpp	2008-08-11 23:49:48.000000000 +0200
+++ boost_1_36_0_beta1/boost/regex/v4/basic_regex_creator.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -36,6 +36,8 @@
 #  pragma warning(disable: 4800)
 #endif
 
+#include <limits.h>
+
 namespace boost{
 
 namespace re_detail{
Only in boost_1_36_0_beta1/boost/regex/v4: basic_regex_creator.hpp.orig
diff -urp boost_1_36_0_beta1-orig/boost/regex/v4/basic_regex.hpp boost_1_36_0_beta1/boost/regex/v4/basic_regex.hpp
--- boost_1_36_0_beta1-orig/boost/regex/v4/basic_regex.hpp	2008-08-11 23:49:48.000000000 +0200
+++ boost_1_36_0_beta1/boost/regex/v4/basic_regex.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -30,6 +30,8 @@
 #pragma warning(pop)
 #endif
 
+#include <limits.h>
+
 namespace boost{
 #ifdef BOOST_MSVC
 #pragma warning(push)
Only in boost_1_36_0_beta1/boost/regex/v4: basic_regex.hpp.orig
diff -urp boost_1_36_0_beta1-orig/boost/regex/v4/basic_regex_parser.hpp boost_1_36_0_beta1/boost/regex/v4/basic_regex_parser.hpp
--- boost_1_36_0_beta1-orig/boost/regex/v4/basic_regex_parser.hpp	2008-08-11 23:49:48.000000000 +0200
+++ boost_1_36_0_beta1/boost/regex/v4/basic_regex_parser.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -30,6 +30,8 @@
 #pragma warning(pop)
 #endif
 
+#include <limits.h>
+
 namespace boost{
 namespace re_detail{
 
Only in boost_1_36_0_beta1/boost/regex/v4: basic_regex_parser.hpp.orig
diff -urp boost_1_36_0_beta1-orig/boost/regex/v4/perl_matcher.hpp boost_1_36_0_beta1/boost/regex/v4/perl_matcher.hpp
--- boost_1_36_0_beta1-orig/boost/regex/v4/perl_matcher.hpp	2008-08-11 23:49:48.000000000 +0200
+++ boost_1_36_0_beta1/boost/regex/v4/perl_matcher.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -30,6 +30,8 @@
 #  pragma warning(disable: 4800)
 #endif
 
+#include <limits.h>
+
 namespace boost{
 namespace re_detail{
 
Only in boost_1_36_0_beta1/boost/regex/v4: perl_matcher.hpp.orig
diff -urp boost_1_36_0_beta1-orig/boost/regex/v4/regex_split.hpp boost_1_36_0_beta1/boost/regex/v4/regex_split.hpp
--- boost_1_36_0_beta1-orig/boost/regex/v4/regex_split.hpp	2008-08-11 23:49:48.000000000 +0200
+++ boost_1_36_0_beta1/boost/regex/v4/regex_split.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -21,6 +21,8 @@
 #ifndef BOOST_REGEX_SPLIT_HPP
 #define BOOST_REGEX_SPLIT_HPP
 
+#include <limits.h>
+
 namespace boost{
 
 #ifdef BOOST_MSVC
Only in boost_1_36_0_beta1/boost/regex/v4: regex_split.hpp.orig
diff -urp boost_1_36_0_beta1-orig/boost/regex/v4/states.hpp boost_1_36_0_beta1/boost/regex/v4/states.hpp
--- boost_1_36_0_beta1-orig/boost/regex/v4/states.hpp	2008-08-11 23:49:48.000000000 +0200
+++ boost_1_36_0_beta1/boost/regex/v4/states.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -30,6 +30,8 @@
 #pragma warning(pop)
 #endif
 
+#include <limits.h>
+
 namespace boost{
 namespace re_detail{
 
Only in boost_1_36_0_beta1/boost/regex/v4: states.hpp.orig
diff -urp boost_1_36_0_beta1-orig/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp boost_1_36_0_beta1/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp
--- boost_1_36_0_beta1-orig/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp	2008-08-11 23:49:45.000000000 +0200
+++ boost_1_36_0_beta1/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp	2008-08-11 23:50:38.000000000 +0200
@@ -16,6 +16,8 @@
 #include <string>
 #include <cstdio>
 #include <cstdarg>
+#include <cstring>
+
 #if defined(BOOST_SPIRIT_DEBUG)
 #include <iostream>
 #endif // defined(BOOST_SPIRIT_DEBUG)
Only in boost_1_36_0_beta1/boost/wave/cpplexer/re2clex: cpp_re2c_lexer.hpp.orig
