*** tools/build/v1/gcc-tools.jam.orig	2005-03-18 09:58:30.227128264 -0600
--- tools/build/v1/gcc-tools.jam	2005-03-18 10:00:20.235025969 -0600
*************** flags gcc FINDLIBS <find-library> ;
*** 352,357 ****
--- 352,358 ----
  flags gcc DLLVERSION <dllversion> ;
  DLLVERSION = $(DLLVERSION[1]) ;
  DLLVERSION ?= $(BOOST_VERSION) ;
+ SONAMEVERSION = 2 ;
  
  flags gcc TARGET_TYPE <target-type> ;
  
*************** rule Link-action
*** 387,393 ****
      if $(DLLVERSION) && $(TARGET_TYPE) in $(SHARED_TYPES) && ( $(OS) = LINUX || $(OS) = OPENBSD ) && ! $(NO_GNU_LN)
      {
          OUTTAG on $(<) = ".$(DLLVERSION)" ;
!         SOTAG on $(<) = ".$(DLLVERSION)" ;
          ACTION_1 on $(<) = "" ;
      }
      else
--- 388,394 ----
      if $(DLLVERSION) && $(TARGET_TYPE) in $(SHARED_TYPES) && ( $(OS) = LINUX || $(OS) = OPENBSD ) && ! $(NO_GNU_LN)
      {
          OUTTAG on $(<) = ".$(DLLVERSION)" ;
!         SOTAG on $(<) = ".$(SONAMEVERSION)" ;
          ACTION_1 on $(<) = "" ;
      }
      else
