*** tools/build/v2/tools/gcc.jam.orig	2008-11-17 16:26:39.000000000 -0800
--- tools/build/v2/tools/gcc.jam	2008-11-17 16:31:08.000000000 -0800
*************** toolset.flags gcc.compile PCH_FILE <pch>
*** 292,298 ****
  
  # Declare flags and action for compilation.
  toolset.flags gcc.compile OPTIONS <optimization>off   : -O0 ;
! toolset.flags gcc.compile OPTIONS <optimization>speed : -O3 ;
  toolset.flags gcc.compile OPTIONS <optimization>space : -Os ;
  
  toolset.flags gcc.compile OPTIONS <inlining>off  : -fno-inline ;
--- 292,298 ----
  
  # Declare flags and action for compilation.
  toolset.flags gcc.compile OPTIONS <optimization>off   : -O0 ;
! toolset.flags gcc.compile OPTIONS <optimization>speed : "_FEDORA_OPT_FLAGS" ;
  toolset.flags gcc.compile OPTIONS <optimization>space : -Os ;
  
  toolset.flags gcc.compile OPTIONS <inlining>off  : -fno-inline ;
