diff -ru boost/spirit/home/classic/iterator/multi_pass.hpp boost/spirit/home/classic/iterator/multi_pass.hpp
--- boost/spirit/home/classic/iterator/multi_pass.hpp	2009-01-09 10:38:36.000000000 +0000
+++ boost/spirit/home/classic/iterator/multi_pass.hpp	2009-01-09 10:39:41.000000000 +0000
@@ -12,7 +12,6 @@
 #include <boost/throw_exception.hpp>
 #include <deque>
 #include <iterator>
-#include <iostream>
 #include <algorithm>    // for std::swap
 #include <exception>    // for std::exception
 #include <boost/limits.hpp>
