*** tools/build/v2/tools/gcc.jam.orig	2007-05-03 08:09:04.000000000 +0200
--- tools/build/v2/tools/gcc.jam	2007-06-26 20:37:44.000000000 +0200
*************** if [ os.name ] != NT && [ os.name ] != O
*** 316,322 ****
      # expected, therefore it has been disabled.
  
      HAVE_SONAME   = "" ;
!     SONAME_OPTION = -h ;
  }
  
  
--- 316,323 ----
      # expected, therefore it has been disabled.
  
      HAVE_SONAME   = "" ;
!     SONAME_OPTION = -soname ;
!     SONAME_VERSION = 3 ;
  }
  
  
*************** rule link.dll ( targets * : sources * : 
*** 631,637 ****
  # Differ from 'link' above only by -shared.
  actions link.dll bind LIBRARIES
  {
!     "$(CONFIG_COMMAND)" -L"$(LINKPATH)" -Wl,$(RPATH_OPTION:E=-R)$(SPACE)-Wl,"$(RPATH)" "$(.IMPLIB-COMMAND)$(<[1])" -o "$(<[-1])" $(HAVE_SONAME)-Wl,$(SONAME_OPTION)$(SPACE)-Wl,$(<[-1]:D=) -shared $(START-GROUP) "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-ST) -l$(FINDLIBS-SA) $(END-GROUP) $(OPTIONS) $(USER_OPTIONS)
  }
  
  # Set up threading support. It's somewhat contrived, so perform it at the end,
--- 632,638 ----
  # Differ from 'link' above only by -shared.
  actions link.dll bind LIBRARIES
  {
!     "$(CONFIG_COMMAND)" -L"$(LINKPATH)" -Wl,$(RPATH_OPTION:E=-R)$(SPACE)-Wl,"$(RPATH)" "$(.IMPLIB-COMMAND)$(<[1])" -o "$(<[-1])" $(HAVE_SONAME)-Wl,$(SONAME_OPTION)$(SPACE)-Wl,$(<[-1]:D=).$(SONAME_VERSION) -shared $(START-GROUP) "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-ST) -l$(FINDLIBS-SA) $(END-GROUP) $(OPTIONS) $(USER_OPTIONS)
  }
  
  # Set up threading support. It's somewhat contrived, so perform it at the end,
