*** tools/build/v1/gcc-tools.jam.orig	2005-03-15 22:41:36.879694255 -0600
--- tools/build/v1/gcc-tools.jam	2005-03-15 22:42:01.449887423 -0600
*************** flags gcc LINKFLAGS <runtime-link>static
*** 60,66 ****
  flags gcc CFLAGS <debug-symbols>on : -g ;
  flags gcc LINKFLAGS <debug-symbols>on : -g ;
  flags gcc CFLAGS <optimization>off : -O0 ;
! flags gcc CFLAGS <optimization>speed : -O3 ;
  
  # Other optimizations we might want for GCC
  # -fforce-mem -fomit-frame-pointer
--- 60,66 ----
  flags gcc CFLAGS <debug-symbols>on : -g ;
  flags gcc LINKFLAGS <debug-symbols>on : -g ;
  flags gcc CFLAGS <optimization>off : -O0 ;
! flags gcc CFLAGS <optimization>speed : -O2 ;
  
  # Other optimizations we might want for GCC
  # -fforce-mem -fomit-frame-pointer
