*** boost/config/compiler/gcc.hpp.orig	2005-03-16 17:25:06.611062784 -0600
--- boost/config/compiler/gcc.hpp	2005-03-16 17:31:00.965341000 -0600
***************
*** 84,91 ****
  #  error "Compiler not configured - please reconfigure"
  #endif
  //
! // last known and checked version is 3.4:
! #if (__GNUC__ > 3) || ((__GNUC__ == 3) && (__GNUC_MINOR__ > 4))
  #  if defined(BOOST_ASSERT_CONFIG)
  #     error "Unknown compiler version - please run the configure tests and report the results"
  #  else
--- 84,91 ----
  #  error "Compiler not configured - please reconfigure"
  #endif
  //
! // last known and checked version is 4.1:
! #if (__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ > 1))
  #  if defined(BOOST_ASSERT_CONFIG)
  #     error "Unknown compiler version - please run the configure tests and report the results"
  #  else
